src/Security/InheritedPermissionFlusher.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace SilverStripe\Security;
+
+use Psr\Log\InvalidArgumentException;
+use SilverStripe\Core\Flushable;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Core\Cache\CacheFlusher;
+
+class InheritedPermissionFlusher extends DataExtension implements Flushable
+{
+    /**
+     * @var CacheFlusher[]
+     */
+    protected $services = [];
+
+    /**
+     * Flush all CacheFlusher services
+     */
+    public static function flush()
+    {
+        singleton(__CLASS__)->flushCache();
+    }
+
+    /**
+     * @param DataObject $owner
+     */
+    public function setOwner($owner)
+    {
+        if (!$owner instanceof Member && !$owner instanceof Group) {
+            throw new InvalidArgumentException(sprintf(
+                '%s can only be applied to %s or %s',
+                __CLASS__,
+                Member::class,
+                Group::class
+            ));
+        }
+
+        parent::setOwner($owner);
+    }
+
+    /**
+     * @param CacheFlusher[]
```

Comment:
Could we please add a `@throws` PHPDoc for the InvalidArgumentException, and maybe return $this for chaining?